### PR TITLE
optimize: modify context package path

### PR DESCRIPTION
**Pull request by 1046102779**

### CODE REVIEW

**Comment by daixiang0** [comment]
Thanks for your contribution, could you fix `govet` issue to make CI happy?

File: pkg/messaging/direct_messaging.go [hunk 1]
```diff
@@ -262,13 +262,20 @@ func (d *directMessaging) invokeRemote(ctx context.Context, appID, namespace, ap
 }
 
 func (d *directMessaging) addDestinationAppIDHeaderToMetadata(appID string, req *invokev1.InvokeMethodRequest) {
-	req.Metadata()[invokev1.DestinationIDHeader] = &internalv1pb.ListStringValue{
+	metadata := req.Metadata()
```
Comment (daixiang0) [inline pkg/messaging/direct_messaging.go:265]: It is another issue, isn't it? I would create another PR to fix it. Also, do not forget add tests.

Comment (1046102779) [reply pkg/messaging/direct_messaging.go:265]: ok, I check and repair it.

**Comment by 1046102779** [comment]
image

it be fixed.

**Comment by daixiang0** [comment]
@1046102779 please make CI happy then we can make it land.

File: pkg/messaging/direct_messaging.go [hunk 2]
```diff
@@ -237,7 +237,7 @@ func (d *directMessaging) setContextSpan(ctx context.Context) context.Context {
 }
 
 func (d *directMessaging) invokeRemote(ctx context.Context, appID, namespace, appAddress string, req *invokev1.InvokeMethodRequest) (*invokev1.InvokeMethodResponse, error) {
-	conn, teardown, err := d.connectionCreatorFn(context.TODO(), appAddress, appID, namespace, false, false, false)
```
Comment (ItalyPaleAle) [inline pkg/messaging/direct_messaging.go:240]: I don't think we should change it here, it was meant as a TODO on purpse IMHO

File: pkg/grpc/proxy/handler_test.go [hunk 3]
```diff
@@ -177,7 +185,10 @@ func (s *ProxyHappySuite) TestDirectorErrorIsPropagated() {
 }
 
 func (s *ProxyHappySuite) TestPingStream_FullDuplexWorks() {
-	stream, err := s.testClient.PingStream(s.ctx())
```
Comment (ItalyPaleAle) [inline pkg/grpc/proxy/handler_test.go:180]: Why this change? This reduces the timeout

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]

File: pkg/messaging/direct_messaging.go (hunk 2 above)
Comment (1046102779) [reply pkg/messaging/direct_messaging.go:240]: Ok, i change it back.

Comment (1046102779) [reply pkg/messaging/direct_messaging.go:240]: ok . i change it back.

File: pkg/grpc/proxy/handler_test.go (hunk 3 above)
Comment (1046102779) [reply pkg/grpc/proxy/handler_test.go:180]: I removed the following method, and then used the `context. WithTimeout` method directly. It is not appropriate to put ctx in the `ProxyHappySuite` object, it should be placed in a entire request.

```golang
func (s *ProxyHappySuite) ctx() context.Context {
```

**Comment by daixiang0** [comment]
@1046102779 could you rebase main branch?

**Review by ItalyPaleAle — APPROVED** [review]

**Review by daixiang0 — APPROVED** [review]
